src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java

```diff
@@ -0,0 +1,512 @@
+        else {
+            while (iterator.hasNext()) {
+                final JsonNode recordJsonNode = iterator.next();
+                dedupAndFetchData(task, recordJsonNode, knownIDs, schema, pageBuilder, startTime);
+
+                // in preview only need to hit one
+                if (Exec.isPreview()) {
+                    break;
+                }
+            }
+        }
+    }
+
+    @VisibleForTesting
+    protected void dedupAndFetchData(final PluginTask task, final JsonNode recordJsonNode, Set<String> knownIDs,
+                                     final Schema schema, final PageBuilder pageBuilder, long startTime)
+    {
+        if (task.getDedup()) {
+            String recordID = recordJsonNode.get(ZendeskConstants.Field.ID).asText();
+            if (!knownIDs.contains(recordID) && !isUpdatedBySystem(recordJsonNode, startTime)) {
+                fetchData(recordJsonNode, task, schema, pageBuilder);
+                knownIDs.add(recordID);
+                count.incrementAndGet();
+            }
+        }
+        else {
+            fetchData(recordJsonNode, task, schema, pageBuilder);
+            count.incrementAndGet();
+        }
+    }
+
+    private JsonNode buildColumns(final PluginTask task)
+    {
+        JsonNode jsonNode = getZendeskSupportAPIService(task).getData("", 0, true, 0);
+
+        String targetName = task.getTarget().getJsonName();
+
+        if (jsonNode.get(targetName) != null && jsonNode.get(targetName).isArray()) {
+            return addAllColumnsToSchema(jsonNode, task.getTarget(), task.getIncludes());
+        }
+        else {
+            throw new ConfigException("Could not guess schema due to empty data set");
+        }
+    }
+
+    private final Pattern idPattern = Pattern.compile(ZendeskConstants.Regex.ID);
+    private JsonNode addAllColumnsToSchema(final JsonNode jsonNode, final Target target, final List<String> includes)
+    {
+        final JsonNode sample = createSamples(jsonNode, target);
+        final Buffer bufferSample = Buffer.copyOf(sample.toString().getBytes(StandardCharsets.UTF_8));
+        final JsonNode columns = Exec.getInjector().getInstance(GuessExecutor.class)
+                .guessParserConfig(bufferSample, Exec.newConfigSource(), createGuessConfig())
+                .getObjectNode().get("columns");
+
+        final Iterator<JsonNode> ite = columns.elements();
+        while (ite.hasNext()) {
+            final ObjectNode entry = (ObjectNode) ite.next();
+            final String name = entry.get("name").asText();
+            final String type = entry.get("type").asText();
+
+            if (name.equals("id")) {
+                if (!type.equals(Types.LONG.getName())) {
+                    entry.put("type", Types.LONG.getName());
+                }
+            }
+            else if (idPattern.matcher(name).matches()) {
+                entry.put("type", Types.STRING.getName());
+            }
+
+            if (type.equals(Types.TIMESTAMP.getName())) {
```

Comment:
what if `name` = `id` and `type` = `timestamp`? does it mean there are 2 entries will be put with `type` = `long` and `format` = `RUBY_TIMESTAMP_FORMAT`?

Reply:
It is still there, the removal is added **BEFORE** put which is useless and not different, please refactor this part for readability.